src/towncrier/newsfragments/676.feature.rst

```diff
@@ -0,0 +1 @@
+new `--staged` parameter for the `check` to included staged files as part of confirming a news file exists
```

Comment:
I'd be nice if this explained why this functionality is useful. Perhaps:

```suggestion
The `towncrier check` command now has a `--staged` flag to inspect the files staged for commit when checking for a news fragment: useful in a pre-commit hook
```

But feel free to wordsmith as you like!